src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java

```diff
@@ -149,22 +146,74 @@ public static DetailAST getAnnotation(final DetailAST ast,
                     "the annotation is empty or spaces");
         }
 
+        return getAnnotation(ast, (annotationNode) -> {
+            final DetailAST firstChild = annotationNode.findFirstToken(TokenTypes.AT);
+            final String name =
+                    FullIdent.createFullIdent(firstChild.getNextSibling()).getText();
+            return annotation.equals(name);
+        });
+    }
+
+    /**
+     * Checks if the given AST element is annotated with any of the specified annotations.
+     *
+     * <p>
+     * This method accepts both simple and fully-qualified names,
+     * e.g. "Override" will match both java.lang.Override and Override.
+     * </p>
+     *
+     * @param ast The type or method definition.
+     * @param annotations A collection of annotations to look for.
+     * @return {@code true} if the given AST element is annotated with at least one of the specified annotations; {@code false} otherwise.
+     */
+    public static boolean containsAnnotation(DetailAST ast, List<String> annotations) {
+        if (ast == null) {
+            throw new IllegalArgumentException(THE_AST_IS_NULL);
+        }
+
+        boolean result = false;
+
+        if (annotations != null && !annotations.isEmpty()) {
+            result = getAnnotation(ast, (annotationNode) -> {
```

Comment:
Why is this lambda's logic so much different then `getAnnotation`'s lambda?
I would think the difference should only be with the return (equals versus contains).

Reply:
Very well spotted and this is what I want to discuss.
The "allowedAnnotations" of JavadocMethodType (which I reused in the first place) works with both "java.lang.Override" and "Override". So if you specify that your allowedAnnotations is "Override", it will happily ignore methods marked with "@override" but also their FQN "@java.lang.Override".

The other method however, which I don't know where it's used, is clearly stating in its documentation that "Override" and "java.lang.Override" are different things and require an explicit second method call.

I want your advice on how to proceed here. I don't like the difference in semantics. Obviously, we can't break backwards compatibility with how JavadocMethodCheck works.

So we need both styles of matching. The strict style, which demands that you specify the annotation as it is defined. The loose style, which matches "Override" and "java.lang.Override", which was used in JavadocMethod (which is why that code was different) and which is the one I'm trying to use in JavadocType.

I hope this makes sense :-)

Reply:
@romani ping

Reply:
@rnveach , 
This Check is not javadoc-ANTLR based, and need to be redo completely.
So right now all such Checks (regexp based) are supported by community in matter of no big damage, but such Checks are skipped from our usual demand for quality.

So if it works a bit better than smth similar close by, it is ok for now.

Reply:
fixed